test/Microsoft.AspNetCore.Mvc.Core.Test/FileStreamResultTest.cs

```diff
@@ -45,11 +48,127 @@ public void Constructor_SetsContentTypeAndParameters()
             var result = new FileStreamResult(stream, contentType);
 
             // Assert
+            Assert.False(result.EnableRangeProcessing);
             Assert.Equal(stream, result.FileStream);
             MediaTypeAssert.Equal(expectedMediaType, result.ContentType);
         }
 
         [Fact]
+        public void Constructor_SetsRangeProcessingParameters()
+        {
+            // Arrange
+            var stream = Stream.Null;
+            var contentType = "text/plain";
+            var expectedMediaType = contentType;
+            var lastModified = new DateTimeOffset();
+            var entityTag = new EntityTagHeaderValue("\"Etag\"");
+
+            // Act
+            var result = new FileStreamResult(
+                fileStream: stream,
+                contentType: new MediaTypeHeaderValue(contentType),
+                enableRangeProcessing: true,
+                lastModified: lastModified,
+                entityTag: entityTag);
+
+            // Assert
+            Assert.True(result.EnableRangeProcessing);
+            Assert.Equal(lastModified, result.LastModified);
+            Assert.Equal(entityTag, result.EntityTag);
+            MediaTypeAssert.Equal(expectedMediaType, result.ContentType);
+        }
+
+        [Theory]
+        [InlineData(0, 4, "Hello", 5)]
+        [InlineData(6, 10, "World", 5)]
+        public async Task WriteFileAsync_WritesRangeRequested(long start, long end, string expectedString, long contentLength)
+        {
+            // Arrange            
+            var contentType = "text/plain";
+            var expectedMediaType = contentType;
+            var lastModified = new DateTimeOffset();
+            var entityTag = new EntityTagHeaderValue("\"Etag\"");
+            var byteArray = Encoding.ASCII.GetBytes("Hello World");
+            var readStream = new MemoryStream(byteArray);
+
+            var result = new FileStreamResult(
+                fileStream: readStream,
+                contentType: new MediaTypeHeaderValue(contentType),
+                enableRangeProcessing: true,
+                lastModified: lastModified,
+                entityTag: entityTag);
+
+            var httpContext = GetHttpContext();
+            httpContext.Request.GetTypedHeaders().Range = new RangeHeaderValue(start, end);
```

Comment:
Can you add a test where the request does not have this range header and the action says `enableRangeProcessing` to `true`?